source/slang/slang-check-decl.cpp

```diff
@@ -6074,11 +6074,9 @@ namespace Slang
 
             if (this->getOptionSet().getBoolOption(CompilerOptionName::ZeroInitialize))
             {
-                // Force add IDefaultInitializableType to any struct missing `IDefaultInitializableType`.
-                // This requires that we check for transitive relationships of all struct->interface, 
-                // struct->interface->interface..., relationship 
+                // Force add IDefaultInitializableType to any struct missing (transitively) `IDefaultInitializableType`.
                 auto* defaultInitializableType = m_astBuilder->getDefaultInitializableType();
-                if(isSubtype(DeclRefType::create(m_astBuilder, decl), defaultInitializableType))
+                if(isSubtype(DeclRefType::create(m_astBuilder, decl), defaultInitializableType), true)
```

Comment:
I am confused here. It seems that this code is trying to test if `this` is a sub type of `IDefaultIinitializable`, and if it is, and we set `-zero-initialize` flag, then we create an inheritance decl on the struct to make it conform to `IDefaultInitializable`.

Why is this needed? If a type transitively inherits `IDefaultInitializable`, then it should also be default initializable without a direct inheritance decl.

Reply:
`                if(isSubtype(DeclRefType::create(m_astBuilder, decl), defaultInitializableType), true)` -->                 if(!isSubtype(DeclRefType::create(m_astBuilder, decl), defaultInitializableType), true)

Reply:
I will also add tests to hopefully not let this type of regression go unnoticed in the future